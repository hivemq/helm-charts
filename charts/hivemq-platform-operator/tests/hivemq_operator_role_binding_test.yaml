suite: HiveMQ Platform Operator - RoleBinding tests
templates:
  - bindings.yml
release:
  name: operator
  namespace: hivemq-platform-operator-namespace
chart:
  version: 0.0.1
  appVersion: 1.0.0
tests:

  - it: with default values, default RoleBinding created
    documentIndex: 0 # FIXME: would be nice to have this value set at the test suite level. See https://github.com/helm-unittest/helm-unittest/issues/480
    asserts:
      - containsDocument:
          apiVersion: rbac.authorization.k8s.io/v1
          kind: RoleBinding
          name: hivemq-platform-operator-view-binding-operator
      - isSubset:
          path: roleRef
          content:
            apiGroup: rbac.authorization.k8s.io
            kind: ClusterRole
            name: view
      - contains:
          path: subjects
          content:
            kind: ServiceAccount
            name: hivemq-platform-operator-operator
            namespace: hivemq-platform-operator-namespace

  - it: with default values, default labels created
    documentIndex: 0
    asserts:
      - isSubset:
          path: metadata.labels
          content:
            helm.sh/chart: hivemq-platform-operator-0.0.1
            app.kubernetes.io/name: hivemq-platform-operator
            app.kubernetes.io/instance: operator
            app.kubernetes.io/version: 1.0.0
            app.kubernetes.io/managed-by: Helm

  - it: with long Helm release name, RoleBinding name truncated
    release:
      name: a-very-long-hivemq-platform-operator-release-name
    documentIndex: 0
    asserts:
      - equal:
          path: metadata.name
          value: hivemq-platform-operator-view-binding-a-very-long-hivemq-platfo

  - it: with disabled ServiceAccount creation, no RoleBinding created
    set:
      serviceAccount.create: false
    asserts:
      - notFailedTemplate: {}
      - hasDocuments:
          count: 0

  - it: with custom ServiceAccount, then custom ServiceAccount used as subject
    set:
      serviceAccount.name: foobar
    documentIndex: 0
    asserts:
      - contains:
          path: subjects
          content:
            kind: ServiceAccount
            name: foobar
            namespace: hivemq-platform-operator-namespace
