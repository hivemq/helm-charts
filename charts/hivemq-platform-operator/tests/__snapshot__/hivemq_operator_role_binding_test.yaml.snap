with default values:
  1: |
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: view
  2: |
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: hivemq-platform-operator-role-operator
  3: |
    - kind: ServiceAccount
      name: hivemq-platform-operator-operator
      namespace: hivemq-platform-operator-namespace
  4: |
    - kind: ServiceAccount
      name: hivemq-platform-operator-operator
      namespace: hivemq-platform-operator-namespace
with namespaces:
  1: |
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: view
  2: |
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: hivemq-platform-operator-role-operator
  3: |
    apiGroup: rbac.authorization.k8s.io
    kind: Role
    name: hivemq-platform-operator-role-operator
  4: |
    apiGroup: rbac.authorization.k8s.io
    kind: Role
    name: hivemq-platform-operator-role-operator
  5: |
    - kind: ServiceAccount
      name: hivemq-platform-operator-operator
      namespace: hivemq-platform-operator-namespace
  6: |
    - kind: ServiceAccount
      name: hivemq-platform-operator-operator
      namespace: hivemq-platform-operator-namespace
  7: |
    - kind: ServiceAccount
      name: hivemq-platform-operator-operator
      namespace: hivemq-platform-operator-namespace
  8: |
    - kind: ServiceAccount
      name: hivemq-platform-operator-operator
      namespace: hivemq-platform-operator-namespace
